distribution.md

```diff
@@ -228,3 +230,89 @@ The Run Image SHOULD contain the following Labels on the image configuration:
 Where,
 
 `<Target ID>` is an identifier specified on the runtime image that MAY be used to apply target-specific logic.
+
+### Builder
+
+The following specifies the artifact format for a buildpacks builder.
+
+A builder is an OCI Image that provides a distributable build environment.
+
+A platform supporting builders SHOULD allow users to configure the build environment for a provided builder.
+
+#### General Requirements
+
+The builder image MUST contain an implementation of the [lifecycle](#lifecycle), and a [build-time](#build-image) environment, and MAY contain [buildpacks](#buildpackage). Platforms SHOULD use builders to ease the build process.
+
+#### Filesystem
+
+A builder MUST have the following directories/files:
+
+* `/cnb/order.toml` &rarr; As defined in the [platform specification](https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml)
+
+#### Environment Variables
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified environment variables set on the original build image, as specified in the Build Image specifications.
+
+The following environment variables MAY be set on the builder (through the image config's `Env` field):
+
+| Env Variable             | Description                                                                       | Default
+| --------                 | --------                                                                          |-----------
+| `CNB_APP_DIR`            | Application directory of the build environment                                    | `/workspace`
+| `CNB_LAYERS_DIR`         | The directory to create and store `layers` in the build environment               | `/layers`
+| `CNB_PLATFORM_DIR`       | The directory to create and store platform focused files in the build environment | `/platform`
+| `SERVICE_BINDING_ROOT`   | The directory where services are bound                                            | -
+
+
+#### Labels
+
+A builder MUST be an extension of a Build Image, and MUST retain all the specified Labels set on the original build image, as specified in the Build Image specifications.
+
+A builder image MUST contain an implementation of the [lifecycle](#lifecycle), and MUST retain all the specified Labels set on the original Lifecycle image, as specified in the lifecycle distribution specifications.
+
+A builder image MAY contain [buildpacks](#buildpackage), and MAY retain all the specified Labels set on the original buildpackage, as specified in the buildpackage specifications.
```

Comment:
I think this section needs some adjutment. Multiple buildpackages will have conflicting (non-compatible) `io.buildpacks.buildpackage.metadata` labels and any `io.buildpacks.buildpackage.layers` labels will need to be merged.

Reply:
I can think of two options:

1. Make the labels unique per buildpack (ie. `io.buildpacks.buildpackage.<id>.<version>.metadata` and `io.buildpacks.buildpackage.<id>.<version>.layers`).
    - Drawbacks: 
        - Predictability and discoverability. Usage would require a more complex glob search in all labels.
        - Duplication. Potential duplication of data when a buildpack is present in multiple meta-buildpacks.
2. `io.buildpacks.buildpackage.metadata` and `io.buildpacks.buildpackage.layers` must be unset and a new label `io.buildpacks.builder.layers` must be created merging `io.buildpacks.buildpackage.layers`.
    - Drawbacks:
        - Additional processing of labels.
        - Can't "duck-type" a builder as a buildpackage. (Maybe that's a good thing).


I'm personally leaning towards option 2. What are your thoughts? Anything I'm overlooking?

Reply:
I think option 3 is to stick with the existing `io.buildpacks.buildpack.layers` merged label. Is there a reason we don't like that option?

Reply:
Did you mean `io.buildpacks.buildpackages.layers` or are you proposing renaming what is already in this branch of the spec?

That aside, I see what you mean. Option 3 is a merge and update of the existing label. Your description doesn't  account for the _other_ buildpackage label `...metadata`. I still think that should be unset.

Personally, I don't have a preference between option 2 or 3+. 

If I were to summarize, what do you think of:

1. Rename `io.buildpacks.builpackages.layers` => `io.buildpacks.builpack.layers`
2. Rename `io.buildpacks.builpackages.metadata` => `io.buildpacks.builpack.metadata`
3. Merge all `io.buildpacks.buildpack.layers` of any buildpackages into `io.buildpacks.builpack.layers` when a builder is built
4. `io.buildpacks.buildpack.metadata` should not be present on a builder image.

_The rename would also align better with `io.buildpacks.buildpack.order` below._

Reply:
I like option 3+

While we would need to rename `io.buildpacks.buildpackages.layers` in this branch of the spec with conveniently **would not** need to rename the label (`io.buildpacks.builpack.layers`) that we are currently applying to buildpacks and builders

Reply:
Resolved in 35e5cce6acb83cfa018f61dc0bc90019375b8702